web/src/components/profile/ProfileNaicsCode.tsx

```diff
@@ -27,13 +29,34 @@ export const ProfileNaicsCode = (): ReactElement => {
     return `/tasks/${urlSlug}`;
   }, [roadmap]);
 
+  const lockField = (): boolean => {
```

Comment:
i would prefer to name this something like `shouldLockField`, which clearly communicates that it returns a boolean

Whereas `lockField` sounds like an action, ie, invoking this function will perform the action to lock the field